incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -879,9 +874,13 @@ private KafkaWillProxy(
             this.replyId = supplyReplyId.applyAsLong(initialId);
             this.willFetchers = new Object2ObjectHashMap<>();
             this.partitions = new IntHashSet();
+            this.reconnectAt = signaler.signalAt(
+                currentTimeMillis(),
+                SIGNAL_CONNECT_WILL_STREAM,
+                this::onSignalConnectWillStream);
```

Comment:
This behavior should be moved out of the constructor to a separate method, as it is not really part of the object construction.

Suggest calling it something like `doKafkaBeginAt` and pass `currentTimeMillis()` as a parameter.